FreeRTOS/FreeRTOS
# AutoRelease + Header Checker Upgrades

Description
-----------

- add `requirements.txt` for easy installation of dependencies `pip -r requirements.txt`
    - strictly depend on python libs so `requirements.txt` can be sufficient
        - Replaced the Git API to use gitpython for local edits/commits etc.
        - Replaced zipping utility with python lib
- refactored/improved some of the older scripts ('versioning.py', 'release.py')
- Add mechanism for undoing an entire flow
- git org should be required input
- Add mechanism for baselining on certain commit, and creating detached HEAD tags
- Version number updates need to update header check ref for kernel and fr/fr
    - decoupled kernel and core configs so they live in their respective repos. Need for next bullet
        - See https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/236 
- Git auth is more flexible for `release.py`. Needed to run in git action environment
- Created git action that wraps script
- Faster
- Added fix to header checker

Test Steps
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
